goss-testing/tests/common/goss-network-interfaces.yaml

```diff
@@ -24,12 +24,17 @@
 # Variable set: network_interfaces
 command:
 {{range $interface := index .Vars "network_interfaces"}}
-  interface_{{$interface}}:
+interface_{{$interface}}:
     title: Interface '{{$interface}}' Has VLAN
     meta:
       desc: Ensures VLAN was created for network interface '{{$interface}}'.
       sev: 0
-    exec: 'ip link show dev {{$interface}}'
+    exec: |-
+        if ! eval 'ip link show dev {{$interface}}' 1>/dev/null; then
+          if [[ "{{$interface}}" == "bond0.can0" ]]; then
+            echo "{{$interface}}is optional...skipping"
+          fi
+        fi
     exit-status: 0
```

Comment:
The above script will always exit 0, I think. Or am I missing something? I don't see how this test would fail.

Reply:
Like with the other PR, though, if you just want to skip the test for can0, using goss conditional execution could be simpler:
```
command:
{{range $interface := index .Vars "network_interfaces"}}
    {{if ne {{$interface}} "bond0.can0"}}
        interface_{{$interface}}_has_vlan:
            title: Interface '{{$interface}}' Has VLAN
            meta:
                desc: Ensures VLAN was created for network interface '{{$interface}}'.
                sev: 0
            exec: ip link show dev {{$interface}}
            exit-status: 0
            timeout: 10000
            skip: false
    {{end}}
{{end}}
```

Reply:
Fixed up.  Thanks!

Reply:
Actually, I was just thinking this will always skip the can.  If it is there, we do want to check it... we might need to re-think this a bit

Reply:
OK I updated the logic here to fail gracefully if can0 is not there:
```
    {{if eq $interface "bond0.can0"}}
    # it's ok if the bond0.can0 fails since it is optional
    exec: "if ! eval ip -4 a s dev {{$interface}} | awk '/inet / {print $2}'; then : ; fi"
    {{else}}
    exec: "ip -4 a s dev {{$interface}} | awk '/inet / {print $2}'"
    {{end}}
```